core/src/main/kotlin/com/alecstrong/sql/psi/core/psi/mixins/WithClauseMixin.kt

```diff
@@ -1,17 +1,24 @@
 package com.alecstrong.sql.psi.core.psi.mixins
 
 import com.alecstrong.sql.psi.core.SqlAnnotationHolder
+import com.alecstrong.sql.psi.core.postgresql.psi.PostgreSqlReturningClause
 import com.alecstrong.sql.psi.core.psi.QueryElement.QueryResult
 import com.alecstrong.sql.psi.core.psi.SqlCompositeElementImpl
+import com.alecstrong.sql.psi.core.psi.SqlCompoundSelectStmt
 import com.alecstrong.sql.psi.core.psi.SqlWithClause
 import com.intellij.lang.ASTNode
+import com.intellij.psi.util.PsiTreeUtil
 
 internal abstract class WithClauseMixin(
   node: ASTNode
 ) : SqlCompositeElementImpl(node),
     SqlWithClause {
   override fun annotate(annotationHolder: SqlAnnotationHolder) {
-    cteTableNameList.zip(compoundSelectStmtList)
+    cteTableNameList.zip(withClauseAuxiliaryStmtList)
+        .mapNotNull { (name, withClauseAuxiliaryStmt) ->
+          PsiTreeUtil.findChildOfAnyType(withClauseAuxiliaryStmt, SqlCompoundSelectStmt::class.java, PostgreSqlReturningClause::class.java)
```

Comment:
and here

Reply:
Fixed in https://github.com/AlecStrong/sql-psi/pull/170/commits/0647c327e733e690fba8344a4bc4c244e24f7dc9.